kubernetes/endpoint_test.go

```diff
@@ -0,0 +1,162 @@
+package kubernetes
+
+import (
+	"testing"
+)
+
+func TestConnection(t *testing.T) {
+	t.Run("Connection bitmask handles Unreachable correctly", func(t *testing.T) {
+		exampleConnection := Connection(0)
+		if exampleConnection != Unreachable {
+			t.Errorf("zero-value Connection should be unreachable, got %s", exampleConnection)
+		}
+		// make sure no weird shenanigans happen with double Unreachable
+		exampleConnection.ClearMethod(Unreachable)
+		if exampleConnection != Unreachable {
+			t.Errorf("zero-value Connection should be unreachable, got %s", exampleConnection)
+		}
+		// note: Don't do this, use SetUnreachable
+		exampleConnection.AddMethod(Unreachable)
+		if exampleConnection != Unreachable {
+			t.Errorf("zero-value Connection should be unreachable, got %s", exampleConnection)
+		}
+
+		exampleConnection.SetUnreachable()
+		if exampleConnection != Unreachable {
+			t.Errorf("zero-value Connection should be unreachable, got %s", exampleConnection)
+		}
+	})
+	t.Run("Connection bitmask handles Proxy correctly", func(t *testing.T) {
+		exampleConnection := Connection(0)
+		exampleConnection.AddMethod(Proxy)
+
+		if !exampleConnection.hasMethod(Proxy) {
+			t.Errorf("expected Proxy, got %s", exampleConnection)
+		}
+		exampleConnection.ClearMethod(Proxy)
+		if exampleConnection != Unreachable {
+			t.Errorf("cleared connection should be unreachable, got %s", exampleConnection)
+		}
+	})
+	t.Run("Connection bitmask handles Direct correctly", func(t *testing.T) {
+		exampleConnection := Connection(0)
+		exampleConnection.AddMethod(Direct)
+
+		if !exampleConnection.hasMethod(Direct) {
+			t.Errorf("expected Direct, got %s", exampleConnection)
+		}
+		exampleConnection.ClearMethod(Direct)
+		if exampleConnection != Unreachable {
+			t.Errorf("cleared connection should be unreachable, got %s", exampleConnection)
+		}
+	})
+
+	t.Run("Connection bitmask handles multiple flags correctly", func(t *testing.T) {
+		exampleConnection := Connection(0)
+		exampleConnection.AddMethod(Direct)
+		exampleConnection.AddMethod(Proxy)
+
+		// should have both methods set
+		if !exampleConnection.hasMethod(Direct) {
+			t.Errorf("expected Direct, got %s", exampleConnection)
+		}
+		if !exampleConnection.hasMethod(Proxy) {
+			t.Errorf("expected Proxy, got %s", exampleConnection)
+		}
```

Comment:
Maybe worth also showing in this sequence of tests that the following also works:

`!exampleConnection.hasMethod(Direct | Proxy)`